revocation/revocation.go

```diff
@@ -0,0 +1,122 @@
+// Package Revocation provides methods for checking the revocation status of a certificate chain
+package revocation
+
+import (
+	"crypto/x509"
+	"net/http"
+	"time"
+
+	"github.com/notaryproject/notation-core-go/revocation/ocsp"
+)
+
+// Revocation is an interface that specifies methods used for revocation checking
+type Revocation interface {
+	Validate(certChain []*x509.Certificate, signingTime time.Time) error
+	OCSPStatus(certChain []*x509.Certificate, signingTime time.Time) error
+	SetMergeErrorsFunction(mergeErrorsFunc func(errors []error) error)
+}
+
+// revocation is an internal struct used for revocation checking
+type revocation struct {
+	httpClient      *http.Client
+	mergeErrorsFunc func(errors []error) error
+}
+
+// New constructs a revocation object and substitutes default values for any that are passed as nil
+func New(httpClient *http.Client) Revocation {
```

Comment:
This was in my original design. It was removed because we decided to use the timeout from a client rather than having a specific timeout option for each type of revocation. I don't see any option specific to OCSP that would differ from CRL revocation, so I think it doesn't make sense to name it related to OCSP. If we just have an options structure similar to my original design, then we can always add more to it if needed in the future without breaking functionality.

Also, if we remove MergeErrorsFunc from the revocation checking (by returning the entire list of errors), then what would be in this options struct for now? I can't think of any other user options that would be needed (other than the client, which I guess we could move in there)